### PR TITLE
Provide migration hook & migrate jobs config to database

### PR DESCRIPTION
**Pull request by yhabteab**
## Blocked by

- #194

### CODE REVIEW

File: library/X509/ProvidedHook/DbMigration.php
Comment (sukhwinder33445) [inline library/X509/ProvidedHook/DbMigration.php:58]: The condition `$columnType !== 'bigint(20) unsigned'` only matches in case schema version is  `< 1.2.0`.  Otherwise `$this->version` is still not initialised.

**Review by sukhwinder33445 — CHANGES_REQUESTED** [review]

File: library/X509/ProvidedHook/DbMigration.php
Comment (nilmerg) [inline library/X509/ProvidedHook/DbMigration.php:83]: is this obsolete?

**Review by nilmerg — CHANGES_REQUESTED** [review]
* Please rebase
* I believe this is not yet updated, since php migrations were dropped, right?

File: library/X509/ProvidedHook/DbMigration.php
Comment (yhabteab) [reply library/X509/ProvidedHook/DbMigration.php:83]: No, this is not obsolete!

**Comment by yhabteab** [comment]
> Please rebase

It's rebased.

> I believe this is not yet updated, since php migrations were dropped, right?

I don't know how you came to that conclusion. This PR is based on another not yet merged PR, in case you haven't noticed.

**Comment by nilmerg** [comment]
> It's rebased.

But there are no checks performed by GitHub :thinking: 

> I don't know how you came to that conclusion.

Well, doesn't the migrate command not skip the call to `migrateJobs` if v2.12 is in use?

**Review by nilmerg — COMMENTED** [review]
The documentation doesn't really emphasize now that the migrate command needs to run in any case. Please adjust that somewhat.

Otherwise looks okay, though my final approve only follows once the checks ran. (after the referenced pr has been merged)

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — DISMISSED** [review]

**Review by nilmerg — APPROVED** [review]

**Comment by yhabteab** [comment]
Just rebased!

**Review by nilmerg — APPROVED** [review]

**Pull request merged**
